src/evergreen/api.py

```diff
@@ -1665,7 +1655,7 @@ def __init__(
             api_server, auth, timeout, log_on_error=log_on_error
         )
 
-    @lru_cache(maxsize=CACHE_SIZE)
+    @lru_cache(maxsize=CACHE_SIZE)  # noqa: B019
```

Comment:
Considered unsafe: we need to deprecate CachedEvergreenApi if it can only be implemented with lru_cache that needs to be tied to an instance. Users of this package can pass a requests session that has cached enabled to the regular EvergreenApi class instead.